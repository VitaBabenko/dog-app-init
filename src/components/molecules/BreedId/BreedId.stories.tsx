import type { Meta, StoryObj } from '@storybook/react';
import { BreedId } from '.';

const meta = {
  title: 'Molecules/BreedId',
  component: BreedId
} satisfies Meta<typeof BreedId>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    breedId: '1'
  }
};
