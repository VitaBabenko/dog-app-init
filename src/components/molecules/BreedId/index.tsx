import { Box, Typography, IconButton, Skeleton, styled } from '@mui/material';
import { FC } from 'react';
import { useGetImageByIdQuery } from '../../../services/images';
import { useGetBreedByIdQuery } from '../../../services/breeds';
import { CardItem } from '../../atoms/CardItem';
import { IconDogFoot } from '../../atoms/IconDogFoot';
import { GroupDogFoot } from '../../atoms/GroupDogFoot';

type BreedIdProps = {
  breedId: string | undefined;
};

const StyledAllBox = styled(Box)({
  paddingBottom: '100px'
});

const StyledBox = styled(Box)({
  display: 'flex',
  gap: '100px',
  paddingTop: '50px',
  paddingBottom: '128px'
});

const StyledIconButton = styled(IconButton)({
  position: 'absolute',
  left: '0%'
});

export const BreedId: FC<BreedIdProps> = ({ breedId }) => {
  const id = Number(breedId);

  const { data: breed, isLoading: isLoadingBreed } = useGetBreedByIdQuery({
    id
  });

  console.log(breed);

  const { data: image, isLoading: isLoadingImage } = useGetImageByIdQuery(
    {
      imageId: breed?.reference_image_id
    },
    { skip: !breed?.reference_image_id }
  );

  return (
    <StyledAllBox>
      <StyledBox>
        {isLoadingImage ? (
          <Skeleton variant="rounded" width={485} height={412} />
        ) : (
          <CardItem variant="smallRadiusBorder" borderRadius={0}>
            <img src={image?.url} alt={image?.id} />
          </CardItem>
        )}
        {isLoadingBreed ? (
          <Skeleton variant="rounded" width={485} height={412} />
        ) : (
          <Box>
            <Typography variant="h1" fontSize={40} color="#5F556D">
              {breed?.name}
              <IconButton>
                <IconDogFoot state="primary" />
              </IconButton>{' '}
            </Typography>
            <Typography>
              <Typography component="span">Weight Range:</Typography>
              <Typography component="span">
                {breed?.weight.metric} kg.
              </Typography>
            </Typography>
            <Typography>
              <Typography component="span">Height:</Typography>
              <Typography component="span">
                {breed?.height.metric} sm.
              </Typography>
            </Typography>
            <Typography>
              <Typography component="span">Life span:</Typography>
              <Typography component="span">{breed?.life_span}</Typography>
            </Typography>
            <Typography>
              {breed?.origin && (
                <Typography component="span">Origin:</Typography>
              )}
              <Typography component="span">{breed?.origin}</Typography>
            </Typography>
            <Typography>
              <Typography component="span">Temperament:</Typography>
              <Typography component="span">{breed?.temperament}</Typography>
            </Typography>
          </Box>
        )}
      </StyledBox>
      <StyledIconButton>
        <GroupDogFoot />
      </StyledIconButton>
    </StyledAllBox>
  );
};
