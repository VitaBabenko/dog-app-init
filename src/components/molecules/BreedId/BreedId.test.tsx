import { composeStories } from '@storybook/react';
import * as stories from './BreedId.stories';
import { testSnapshot } from '../../../test/utils/snapshot.utils';

describe('BreedId', () => {
  const { Primary } = composeStories(stories);
  testSnapshot(<Primary />);
});
