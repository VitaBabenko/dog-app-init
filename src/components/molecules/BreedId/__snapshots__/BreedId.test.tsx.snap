// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreedId should match snapshot Component 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.4;
  }

  100% {
    opacity: 1;
  }
}

.emotion-0 {
  padding-bottom: 100px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 100px;
  padding-top: 50px;
  padding-bottom: 128px;
}

.emotion-2 {
  display: block;
  background-color: rgba(95, 85, 109, 0.11);
  height: 1.2em;
  border-radius: 4px;
  -webkit-animation: animation-0 2s ease-in-out 0.5s infinite;
  animation: animation-0 2s ease-in-out 0.5s infinite;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 2.142857142857143rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border: none;
  box-shadow: none;
  position: absolute;
  left: 0%;
}

.emotion-4::-moz-focus-inner {
  border-style: none;
}

.emotion-4.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-4 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-4.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-4:hover {
  background-color: none;
  border: none;
  box-shadow: none;
}

.emotion-4:active,
.emotion-4:focus {
  background-color: none;
  border: none;
  box-shadow: none;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <div
      class="MuiBox-root emotion-1"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse emotion-2"
        style="width: 485px; height: 412px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse emotion-2"
        style="width: 485px; height: 412px;"
      />
    </div>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-4"
      tabindex="0"
      type="button"
    >
      <test-file-stub
        aria-hidden="true"
        classname="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-c6m6ws-MuiSvgIcon-root"
        focusable="false"
      />
    </button>
  </div>
</div>
`;
